src/CacheMiddleware.php

```diff
@@ -115,10 +124,12 @@ public function __invoke(callable $handler)
             if (!isset($this->httpMethods[strtoupper($request->getMethod())])) {
                 // No caching for this method allowed
 
-                return $handler($request, $options)->then(
+                return $handler($request, $options)->then(  
                     function (ResponseInterface $response) use ($request) {
-                        // Invalidate cache after a call of non-safe method on the same URI
-                        $response = $this->invalidateCache($request, $response);
+                        if (!in_array($request->getMethod(), $this->safeMethods)) {
```

Comment:
Can we avoid the `in_array` ? It's not very efficient, perhaps you can use the construction from `$httpMethods`: `['GET' => true, 'HEAD' => true', ...]` and check `isset($this->safeMethods[strtoupper($request->getMethod())])`, WDYT ?

Reply:
I don’t have a problem with that. It would be consistent with the $httpMethods array